src/zenml/integrations/seldon/steps/seldon_deployer.py

```diff
@@ -0,0 +1,194 @@
+    step_env = cast(StepEnvironment, Environment()[STEP_ENVIRONMENT_NAME])
+    pipeline_name = step_env.pipeline_name
+    pipeline_run_id = step_env.pipeline_run_id
+    step_name = step_env.step_name
+
+    def prepare_service_config(model_uri: str) -> SeldonDeploymentConfig:
+        """Prepare the model files for model serving and create and return a
+        Seldon service configuration for the model.
+
+        This function ensures that the model files are in the correct format
+        and file structure required by the Seldon Core server implementation
+        used for model serving.
+
+        Args:
+            model_uri: the URI of the model artifact being served
+
+        Returns:
+            The URL to the model ready for serving.
+        """
+        served_model_uri = os.path.join(
+            context.get_output_artifact_uri(), "seldon"
+        )
+        fileio.makedirs(served_model_uri)
+
+        # TODO [ENG-773]: determine how to formalize how models are organized into
+        #   folders and sub-folders depending on the model type/format and the
+        #   Seldon Core protocol used to serve the model.
+
+        # TODO [MEDIUM]: auto-detect built-in Seldon server implementation
+        #   from the model artifact type
+
+        # TODO [MEDIUM]: validate the model artifact type against the
+        #   supported built-in Seldon server implementations
+        if config.service_config.implementation == "TENSORFLOW_SERVER":
+            # the TensorFlow server expects model artifacts to be
+            # stored in numbered subdirectories, each representing a model
+            # version
+            io_utils.copy_dir(model_uri, os.path.join(served_model_uri, "1"))
+        elif config.service_config.implementation == "SKLEARN_SERVER":
+            # the sklearn server expects model artifacts to be
+            # stored in a file called model.joblib
+            model_uri = os.path.join(model.uri, "model")
+            if not fileio.exists(model.uri):
+                raise RuntimeError(
+                    f"Expected sklearn model artifact was not found at "
+                    f"{model_uri}"
+                )
+            fileio.copy(
+                model_uri, os.path.join(served_model_uri, "model.joblib")
+            )
+        else:
+            # default treatment for all other server implementations is to
+            # simply reuse the model from the artifact store path where it
+            # is originally stored
+            served_model_uri = model_uri
+
+        return SeldonDeploymentConfig(
+            model_uri=served_model_uri,
+            model_name=config.service_config.model_name,
+            implementation=config.service_config.implementation,
+            pipeline_name=pipeline_name,
+            pipeline_run_id=pipeline_run_id,
+            pipeline_step_name=step_name,
+            replicas=config.service_config.replicas,
+            secret_name="seldon-init-container-secret",
```

Comment:
Could it make sense to track this a constant somewhere?

Reply:
You found a bug. This shouldn't be a constant anymore.